src/rules/awaitPromiseRule.ts

```diff
@@ -34,41 +43,43 @@ export class Rule extends Lint.Rules.TypedRule {
 
     public static FAILURE_STRING = "'await' of non-Promise.";
 
-    public applyWithProgram(srcFile: ts.SourceFile, program: ts.Program): Lint.RuleFailure[] {
-        return this.applyWithWalker(new Walker(srcFile, this.getOptions(), program));
+    public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): Lint.RuleFailure[] {
+        const promiseTypes = new Set(["Promise", ...this.ruleArguments.map(String)]);
```

Comment:
I saw `.map(String)` in another rule and thought it to be a good idea. It really makes sure that `promiseTypes` are only `String`s. A [type assertion](https://basarat.gitbooks.io/typescript/docs/types/type-assertion.html) only makes the compiler happy but at runtime another type may occur.